openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -134,24 +179,33 @@ def patch(self, request, username):
         return Response(status=status.HTTP_204_NO_CONTENT)
 
     @staticmethod
-    def update_account(requesting_user, username, update):
-        """Update the account for the given username.
+    def update_account(requesting_user, update, username=None):
+        """Update user account information.
 
         Note:
-            No authorization or permissions checks are done in this method. It is up to the caller
-            of this method to enforce the contract that this method is only called if
-            requesting_user.username == username or requesting_user.is_staff == True.
+            It is up to the caller of this method to enforce the contract that this method is only called
+            by with the user who made the request.
 
         Arguments:
-            requesting_user (User): the user who initiated the request
-            username (string): the username associated with the account to change
-            update (dict): the updated account field values
+            requesting_user (User): The user modifying the account information. Only the user with username
+                'username' has permissions to modify account information.
+            update (dict): The updated account field values.
+            username (string): Optional username specifying which account should be updated. If not specified,
+                `requesting_user.username` is assumed.
 
         Raises:
-            AccountUserNotFound: no user exists with the specified username
+            AccountUserNotFound: `username` was specified, but no user exists with that username.
             AccountUpdateError: the update could not be completed, usually due to validation errors
                 (for example, read-only fields were specified or field values are not legal)
+            AccountNotAuthorized: the requesting_user does not have access to change the account
```

Comment:
I really like that these helper methods are doing authorization and raising AccountNotAuthorized as necessary.